clients/python-pyo3/tensorzero/__init__.py

```diff
@@ -0,0 +1,258 @@
+import json
+import typing as t
+import warnings
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from json import JSONEncoder
+from typing import Any, Dict, List, Optional, Union
+from uuid import UUID
+
+import httpx
+
+from .tensorzero import (
+    AsyncTensorZeroGateway as AsyncTensorZeroGateway,
+)
+from .tensorzero import (
+    TensorZeroGateway as TensorZeroGateway,
+)
+
+
+class ToDictEncoder(JSONEncoder):
+    def default(self, o):
+        return o.to_dict()
+
```

Comment:
why do we need this?

Reply:
This is used by the Rust code when serializing things to JSON (to later be deserialized through serde). I added a comment